src/wp-includes/option.php

```diff
@@ -1280,7 +1352,7 @@ function set_transient( $transient, $value, $expiration = 0 ) {
 		$transient_option  = '_transient_' . $transient;
 
 		if ( false === get_option( $transient_option ) ) {
-			$autoload = 'yes';
+			$autoload = 'yes'; // Consider using null, so default values are selected here.
```

Comment:
I tend to agree here, but I think it's safer to leave this as is for now and tackle separately. The comment you have here makes sense though, so let's keep it for future reference. But we can tackle optimizing transient autoloading in a separate ticket.